src/fromPairs.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Creates a new object from an array of tuples by pairing up first and second elements as {[key]: value}.
+ * If a tuple is not supplied for any element in the array, the element will be ignored
+ * If duplicate keys exist, the tuple with the greatest index in the input array will be preferred.
+ * @param tuples the list of input tuples
+ * @signature
+ *   R.fromPairs(tuples)
+ * @example
+ *   R.fromPairs([['a', 'b'], ['c', 'd']]) // => {a: 'b', c: 'd'}
+ * @category Object
+ */
+export function fromPairs<K extends string | number | symbol, V>(tuples: Array<[K, V]>): Record<K, V>
+
+
+export function fromPairs(tuples: Array<[string | number | symbol, unknown]>) {
+  return tuples.reduce((acc, curr) => curr && curr.length === 2 ? ({
```

Comment:
It actually currently does throw a type error.  This handling is present to prevent errors in case someone ignores the type error.
![image](https://user-images.githubusercontent.com/1848220/95134772-73a64300-0731-11eb-9f1e-01b3366b495c.png)